main/main.c

```diff
@@ -1162,7 +1162,7 @@ static ZEND_COLD void php_error_cb(int type, const char *error_filename, const u
 				syslog(LOG_ALERT, "PHP %s: %s (%s)", error_type_str, buffer, GetCommandLine());
 			}
 #endif
-			spprintf(&log_buffer, 0, "PHP %s:  %s in %s on line %d", error_type_str, buffer, error_filename, error_lineno);
+			spprintf(&log_buffer, 0, "PHP %s:  %s in %s on line %lu", error_type_str, buffer, error_filename, (long unsigned) error_lineno);
```

Comment:
Rather than casting to `unsigned long`, wouldn't it be better to use `PRIu32` here (and in other places)?

Reply:
**You** certainly could ;) as the old saying goes, "if you want something done your way, do it yourself".

According to the C standard, the types `uint32_t` are only required to exist for configurations that have 8, 16 or 32-bit bytes; for any other configuration, those types can't exist. I guess it depends how much work you want to do if the future becomes dependant upon, say, 21-bit bytes (which are most suitable for the non-English speaking/unicode world)...

I'm into writing C code that is as portable as possible and will run on systems dating back to the 80s with a few tweaks if necessary, so you'll see me using `long unsigned` to store a 32-bit (possibly larger) unsigned integer. If that goes against the standard for this project, no big deal. I don't really respect PHP all that much, anyway; I'm only submitting patches to prove to someone that PHP is an inferior product.

Peace!